receiver/hostmetricsreceiver/testdata/config.yaml

```diff
@@ -0,0 +1,23 @@
+receivers:
+  hostmetrics:
+    scrapers:
```

Comment:
Is it necessary to have these under scrapers instead of being top-level?

Reply:
I think it'd be more confusing to not have a header to separate any top level config from the dynamic scraper config? (both for readability and for the code which would have to figure out which keys are top level or not)

Reply:
Yeah, it probably avoids some potential confusion at the expense of nesting. Though in the [example config](https://docs.google.com/document/d/1i-QG-TxkIWVbdx86UM2GXcFvpeCV7zSBsPmMyMkOEBk/edit#heading=h.ac3q70oa20s9) we did without the nesting it's not too confusing I think. @tigrannajaryan has a good eye for config so maybe he can chime in. 

I think your way is the safer approach though. What do you think about making it `collectors` or `collect` to keep with the `collection_interval` naming scheme?

Reply:
From what I see there is `default_collection_interval` at the top-level so we likely want the additional nesting for consistency. Otherwise we would have `default_collection_interval` and (for example) `cpu` as siblings, which does not seem to make sense.

If we _don't_ need `default_collection_interval` then I would prefer to eliminate `scraper` and reduce the nesting. I glanced at the PR and cannot see how `DefaultCollectionInterval` is used in the code (I assume it will be in a future PR?). Is the idea that there is a "default" interval and each scraper can override this? Would it make sense to just hard-code the default and eliminate the config key?

Reply:
Makes sense - I think I'll leave it as is for now though. While we could hardcode `default_collection_interval` (which as you correctly point out will be used in a future PR as the default interval for individual scrapers), there's likely to be other top level config that's needed, e.g. `proc_path` in the example config @jrcamp linked.

@jrcamp - regarding the naming, I originally called these collectors, but ended up changing to scrapers as I thought that was too confusing / broad given the name of this repo is "collector".